opensquirrel/squirrel_ir.py

```diff
@@ -137,6 +137,25 @@ def __init__(self, generator, arguments):
         self.generator = generator
         self.arguments = arguments
 
+    def _check_is_anonymous(self) -> bool:
+        """A check to verify if BlochSphereRotation is a pre-defined single qubit gate.
+
+        Returns:
+            Whether the BlochSphereRotation reflects a gate or not (is an anonymous gate).
+        """
+        if self.arguments is not None:
+            return False
+        from opensquirrel.default_gates import default_gate_set
+
+        for _, gate in enumerate(default_gate_set):
+            gate_args = inspect.signature(gate).parameters.values()
+            if len(list(gate_args)) == 1:
+                output_bloch = gate(self.get_qubit_operands())
+                if np.allclose(output_bloch.axis, self.axis) and np.allclose(output_bloch.phase, self.phase):
+                    Gate.__init__(self, gate, self.get_qubit_operands())
```

Comment:
Does this line have an effect?

Reply:
Even if it did, I find it obscure calling a constructor from a method. I see this should be reserved for calling a base class constructor from a derived class constructor.

Following up on my comment above, we would consider this code suggestion below.

    class Gate()
        def _set_qubit_operands(self, qubit_operands):
            self.qubit_operands = qubit_operands

        def _find_close_default_bloch():
        """Search a default BlochSphereRotation that is close to this gate.
        
        Returns:
            A default BlockSphereRotation if there is any close to this gate, or None otherwise.
        """
        # current _check_is_anonymous implementation,
        # but returning a BlochSphereRotation (true case) or None (false case)

        def is_anonymous():
            return self.arguments is None
    
        def try_name_as_default_bloch():
            default_bloch = self._find_close_default_bloch(self)
            if default_bloch is not None:
                self._set_qubit_operands(default_block.get_qubit_operands())

    # Somewhere in the project
    if gate.is_anonymous():
        gate.try_name_as_default_bloch()